docs/references/config/config.toml.md

```diff
@@ -0,0 +1,1749 @@
+
+### p2p.max_num_outbound_peers
+Maximum number of outbound peers.
+```toml
+max_num_outbound_peers = 10
+```
+
+| Value type          | integer |
+|:--------------------|:--------|
+| **Possible values** | &gt;= 0 |
+
+The [`p2p.max_num_inbound_peers`](#p2pmax_num_inbound_peers) and
+[`p2p.max_num_outbound_peers`](#p2pmax_num_outbound_peers) values together define how many P2P connections the node will
+maintain at maximum capacity, not including list of nodes in the [`p2p.persistent_peers`](#p2ppersistent_peers) setting.
+
+The connections are bidirectional, so any connection can send or receive blocks and other data. The separation into
+inbound and outbound setting only distinguishes the initial setup of the connection: outbound connections are initiated
+by the node while inbound connections are initiated by a remote party.
+
+Nodes on non-routable networks have to set their gateway to port-forward the P2P port for inbound connections to reach
+the node. Outbound connections can be initiated as long as the node has generic Internet access. (Using NAT or methods.)
+
+### p2p.unconditional_peer_ids
+List of node IDs, that are allowed to connect to the node even when connection limits are exceeded.
+```toml
+unconditional_peer_ids = ""
+```
+
+| Value type          | string (comma-separated)         |
+|:--------------------|:---------------------------------|
+| **Possible values** | comma-separated list of node IDs |
+|                     | `""`                             |
+
+If a peer listed in this property requests a connection, it will be accepted, even if the
+[`p2p.max_num_inbound_peers`](#p2pmax_num_inbound_peers) or the
+[`p2p.max_num_outbound_peers`](#p2pmax_num_outbound_peers) are reached.
+
+Contrary to other settings, only the node ID has to be defined here, not the IP:port of the remote node.
+
+### p2p.persistent_peers_max_dial_period
+Maximum pause when redialing a persistent peer.
+```toml
+persistent_peers_max_dial_period = "0s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+If `"0s"` is set, then exponential backoff is applied to re-dial the remote node over and over.
+
+<!--- Todo: add if there are any limitations to the exponential back-off. --->
+
+### p2p.flush_throttle_timeout
+Time to wait before flushing messages out on the connection.
+```toml
+flush_throttle_timeout = "100ms"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0ms"`     |
+
+Write (flush) any buffered data to the connection. The flush is throttled, so if multiple triggers come in within the
+duration, only one flush is executed.
+
+Setting the value to `0ms` is possible, but the behaviour is undefined.
+
+<!--- Todo: trace the code and update what happens if this is set to 0. --->
```

Comment:
This guy: https://github.com/cometbft/cometbft/blob/main/internal/timer/throttle_timer.go

Reply:
And this guy: https://github.com/cometbft/cometbft/blob/main/p2p/conn/connection.go#L559

Reply:
After a send routine, enables this timer to produce an event within flush_throttle_timeout. If set to zero, no sleep time here.

Reply:
You probably should confirm with Ethan.